src/common/tracer.h

```diff
@@ -0,0 +1,31 @@
+#ifndef TRACER_H_
+#define TRACER_H_
+
+#define SIGNED_RIGHT_SHIFT_IS 1
+#define ARITHMETIC_RIGHT_SHIFT 1
+
+#include <arpa/inet.h>
+#include <yaml-cpp/yaml.h>
+
+#include <jaegertracing/Tracer.h>
+
+namespace jaeger_tracing{
+
+  typedef opentracing::Span jspan;
```

Comment:
`unique_ptr<jaeger_tracing::jspan>` unique_ptr which but I m using `auto` to keep it simple
`jaeger_tracing::opentracing::Span` this is raw pointer that is passed around as a pointer to functions, so I have choose to trypedef the second one otherwise the functions will look ugly, and for the first one I m already using auto instead of writing those